processor/src/test/resources/fixtures/org/mapstruct/ap/test/bugs/_1561/Issue1561MapperImpl.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.bugs._1561;
+
+import java.util.stream.Stream;
+import javax.annotation.Generated;
+
+@Generated(
+    value = "org.mapstruct.ap.MappingProcessor",
+    date = "2019-01-28T20:09:22+0100",
+    comments = "version: , compiler: Eclipse JDT (Batch) 1.2.100.v20160418-1457, environment: Java 1.8.0_181 (Oracle Corporation)"
+)
+public class Issue1561MapperImpl implements Issue1561Mapper {
+
+    @Override
+    public Target map(Source source) {
+        if ( source == null ) {
+            return null;
+        }
+
+        Target target = new Target();
+
+        target.setNestedTarget( sourceToNestedTarget( source ) );
+
+        return target;
+    }
+
+    @Override
+    public Source map(Target target) {
+        if ( target == null ) {
+            return null;
+        }
+
+        Source source = new Source();
+
+        Stream<String> nestedTargetProperty = targetNestedTargetProperties( target );
+        if ( nestedTargetProperty != null ) {
+            targetNestedTargetProperties( target ).forEach( source::addProperty );
```

Comment:
I think that the `nestedTargetProperty` should be used instead? See my other comments above